### PR TITLE
catch and log error from shell sync

### PR DESCRIPTION
**Pull request by Nokel81**
Signed-off-by: Sebastian Malton <smalton@mirantis.com>

### CODE REVIEW

File: src/main/index.ts [hunk 1]
```diff
@@ -30,7 +30,11 @@ if (app.commandLine.getSwitchValue("proxy-server") !== "") {
 }
 
 async function main() {
-  shellSync(app.getLocale());
+  try {
+    await shellSync(app.getLocale());
+  } catch (err) {
+    logger.warn("shellSync: ", err);
+  }
```
Comment (ixrock) [inline src/main/index.ts]: Is it expected behaviour of `shell-env`? (throwing error)
Maybe it's better to put it inside `shell-sync.ts`?

File: src/main/shell-sync.ts [hunk 2]
```diff
@@ -11,9 +11,10 @@ interface Env {
  * useful on macos where this always needs to be done.
  * @param locale Should be electron's `app.getLocale()`
  */
-export function shellSync(locale: string) {
+export async function shellSync(locale: string) {
   const { shell } = os.userInfo();
-  const env: Env = JSON.parse(JSON.stringify(shellEnv.sync(shell)))
+  
+  const env: Env = JSON.parse(JSON.stringify(await shellEnv(shell)))
```
Comment (ixrock) [inline src/main/shell-sync.ts]: btw, we have `common/utils/cloneJson.ts` for cloning plain object with JSON.stuff

**Review by ixrock — APPROVED** [review]
LGTM

File: src/main/index.ts (hunk 1 above)
Comment (Nokel81) [reply src/main/index.ts]: Yes, `shell-env` is expected to throw error if something happens. I'll move it to the other file.

**Review by jakolehm — APPROVED** [review]
LGTM. This probably will fix some of the boot issues we have seen.

**Comment by nevalla** [comment]
Why not catching error from shell-sync and using empty object as `env` on error? Then we could keep this:

```typescript
    if (!env.LANG) {
      // the LANG env var expects an underscore instead of electron's dash
      env.LANG = `${app.getLocale().replace('-', '_')}.UTF-8`;
    } else if (!env.LANG.endsWith(".UTF-8")) {
      env.LANG += ".UTF-8"
    }

    // Overwrite PATH on darwin
    if (process.env.NODE_ENV === "production" && process.platform === "darwin") {
      process.env["PATH"] = env.PATH
    }

    // The spread operator allows joining of objects. The precedence is last to first.
    process.env = {
      ...env,
      ...process.env,
    };
```

**Review by nevalla — APPROVED** [review]
LGTM

**Pull request merged**